### ISSUE TITLE
Update kubectrl to newer version

### ISSUE THREAD

**Issue by c1freitas** [InVisionApp/kit-deployer#78]
The SRE's are spinning up a cluster running version 1.6.3. The plan is to move all clusters to this version. We will need to update the kubectl and to test against the 1.6.3 cluster. 

@InVisionApp/engineering-velocity

**Comment by chesleybrown** [comment]
![tenor](https://user-images.githubusercontent.com/492166/27540047-48a9ccfe-5a59-11e7-8926-3058e9c48020.gif)

**Comment by chesleybrown** [comment]
Running this on testing for a day or two.

**Comment by chesleybrown** [comment]
I will push this out on Monday. 😀

**Comment by chesleybrown** [comment]
Had to rollback. Did a test deploy and saw this error:

> private-staging-cluster - Error running kubectl.apply('/tmp/kit-deployer/5d917152-8793-4bb3-8046-6472ebb5e96e/private-staging-cluster-testrosie-svc.yaml.json') Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply

**Comment by chesleybrown** [comment]
This is where the error message comes from:

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/apply.go#L69

Looks like it means we need to create our resources using `kubectl create --save-config` (I think?).

```
--save-config=false: If true, the configuration of current object will be saved in its
annotation. This is useful when you want to perform kubectl apply on this object in the future.
```

I'm not sure how we retro-apply this... 🤔

**Comment by c1freitas** [comment]
@wincus We found an issue with this and are looking into a fix.

**Comment by wincus** [comment]
thanks @c1freitas for taking care of it! Hope is not a big time consumer! let me know if there is anything we can help with!

**Comment by c1freitas** [comment]
I did some research/testing on this issue. This is only a warning and the apply command works fine. There is a work around to disable this _warning_ documented as part of this issue:  https://github.com/kubernetes/kubernetes/issues/25241#issuecomment-217599165

This involves patching the resource to include the configuration information.

However, in testing this with v1.7.0 and v1.6.4 against clusters running 1.4.2 and `1.6.1` This is only a warning and running the `apply` command updates the config annotation correctly so that subsequent deployments via `apply` do not generate this warning.

**Comment by brianhall** [comment]
Great work, @c1freitas. That's great news.

**Comment by brianhall** [comment]
The SRE team is getting the v7 namespace ready for production starting next week. Before they can do that, they'll need this change to get in. Sounds like we're good to test, just an FYI.

**Comment by c1freitas** [comment]
I was planning to wait to merge the PR until after fast-rollbacks is out. That could push it to Friday or next week. Is that ok @brianhall ?

**Comment by brianhall** [comment]
Totally (and I should have noted that here). That'll still make it possible for us to merge this in this week, just _after_ fast rollbacks are launched.

**Comment by chesleybrown** [comment]
We have this deployed to testing and will monitor it there to see if the deploy error rates increase over the next few days. If all is good we will release to production.

**Comment by chesleybrown** [comment]
I couldn't go directly by the deploy failure count on testing as it continuously has a lot of failed deploys as a lot of testing environments are not configured yet.

However, I've reviewed the testing logs of kit-server and I don't see any errors related to the `--save-config` flag. So I'm deeming this safe to deploy to staging/production.

**Comment by chesleybrown** [comment]
This is now closed with #92 and InVisionApp/kit-server/pull/172
